pandas/core/arrays/categorical.py

```diff
@@ -2286,42 +2286,24 @@ def isin(self, values) -> npt.NDArray[np.bool_]:
         return algorithms.isin(self.codes, code_values)
 
     def _replace(self, *, to_replace, value, inplace: bool = False):
+        from pandas import (
+            Index,
+            Series,
```

Comment:
no good way to avoid this?  we generally want core.arrays to be "above" index/series/frame in the dependency structure (not a deal-breaker as Categoricla already relies on Index)

Reply:
I originally tried to implement this without using `Series.replace` but it felt like I was rediscovering a bunch of edge cases that `Series.replace` already handles. Using `Series.replace` seemed a lot cleaner, but I can take another look if you want to avoid it.

Reply:
@jbrockmendel - I took another look at this and didn't come up with any new ideas for avoiding it. As you mentioned, Categorical already relies on Index and a few other Categorical methods already import Series for what I assume are similar reasons.

Reply:
gentle ping @jbrockmendel